### PR TITLE
environment variables for terminal customization

### PR DESCRIPTION
**Pull request by gtritchie**
Low-tech way to provide some per-project terminal customization possibilities.

`$RSTUDIO_PROJ_NAME` - will be empty if no project is loaded, or the base project name if a project is loaded

`$RSTUDIO_PROJ_PATH` - will be home directory is no project is loaded, or the path to the project (minus the project file) when a project is loaded

For example, could create a `~/.rstudio-bash` script that creates/reattaches to a per-project tmux session:

```
tmux attach -t rstudio-$RSTUDIO_PROJ_NAME 2> /dev/null || tmux new -s rstudio-$RSTUDIO_PROJ_NAME
```
and load it via:

### CODE REVIEW

**Comment by gtritchie** [comment]
I added $RSTUDIO_SESSION_ID.

File: src/cpp/session/SessionConsoleProcess.cpp [hunk 1]
```diff
@@ -234,7 +235,21 @@ void ConsoleProcess::commonInit()
                            options_.smartTerminal ? core::system::kSmartTerm :
                                                     core::system::kDumbTerm);
 
+      // environment variables that can be used when configuring custom shells
       core::system::setenv(&(options_.environment.get()), "RSTUDIO_TERM", procInfo_->getHandle());
+
+      core::system::setenv(&(options_.environment.get()), "RSTUDIO_PROJ_NAME",
+                           projects::projectContext().file().stem());
+
+      FilePath projPath = core::system::userHomePath();
```
Comment (jmcphers) [inline src/cpp/session/SessionConsoleProcess.cpp]: The user home path isn't necessarily the default working path when no project is open (see http://docs.rstudio.com/ide/server-pro/r-sessions.html#working-directories) -- is that important for your use case here or do you just need a safe fallback?

Comment (gtritchie) [inline src/cpp/session/SessionConsoleProcess.cpp]: Just wanted a safe fallback, but I should make it consistent and use the same configured location.

Was thinking of use case of per-project-specific shell config file, potentially including one for "no project" case. To make this robust in multi-user/multi-session, guess they'd need to string together all these env variables (proj_path / $user-$session-myrc)?

Comment (jmcphers) [inline src/cpp/session/SessionConsoleProcess.cpp]: Not sure I understand the goal of such a file, let's discuss realtime. In the meantime it looks like you've removed this path entirely (possibly in favor of just using a raw session id?) so will merge as-is and we can make future tweaks on master.

**Pull request merged**